mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java

```diff
@@ -124,4 +129,24 @@ public void matchShouldReturnRecipientsWhenAtLeastOneMimeTypeMatch() throws Exce
 
         assertThat(matcher.match(mail)).containsExactlyElementsOf(mail.getRecipients());
     }
+
+    @Test
+    void matchShouldNotFailOnEmptyCharset() throws Exception {
+        matcher.init(FakeMatcherConfig.builder()
+            .matcherName("HasMimeType")
+            .condition("text/html")
+            .build());
+
+        MimeMessage message = mock(MimeMessage.class);
+        when(message.getHeader(CONTENT_TYPE)).thenReturn(new String[] {"text/html; charset="});
```

Comment:
is it a legal value?

Reply:
It is seen on the PROD... I bet we should not fail as technically the mimetype is specified.

Reply:
Recovering from an illegal charset value seems legitimate to me.